### ISSUE TITLE
Desktop game crashes in launcher at the end of sound track

### ISSUE THREAD

**Issue by mymedia2** [00-Evan/shattered-pixel-dungeon#797]
Thank you for the game!

I tried to start the game several times and ran into different errors. I clicked nothing on start screen. The crashes may have related to broken resources.

```
mymedia@barberry:~/код/игры/shattered-pixel-dungeon$ java -jar ./desktop/build/libs/desktop-1.0.2.jar 
[GAME] java.lang.NullPointerException
	at com.jcraft.jorbis.Info.clear(Info.java:114)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.readPCM(OggInputStream.java:425)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:436)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:458)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:474)
	at com.badlogic.gdx.backends.lwjgl3.audio.Ogg$Music.read(Ogg.java:43)
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALMusic.fill(OpenALMusic.java:253)
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALMusic.play(OpenALMusic.java:93)
	at com.watabou.noosa.audio.Music.play(Music.java:169)
	at com.watabou.noosa.audio.Music.access$400(Music.java:36)
	at com.watabou.noosa.audio.Music$1$1.run(Music.java:157)

[GAME] com.badlogic.gdx.utils.GdxRuntimeException: Failure reading Vorbis.
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.getPageAndPacket(OggInputStream.java:180)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.readPCM(OggInputStream.java:293)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:436)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:458)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:474)
	at com.badlogic.gdx.backends.lwjgl3.audio.Ogg$Music.read(Ogg.java:43)
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALMusic.fill(OpenALMusic.java:253)
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALMusic.update(OpenALMusic.java:237)
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALLwjgl3Audio.update(OpenALLwjgl3Audio.java:253)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.loop(Lwjgl3Application.java:132)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.<init>(Lwjgl3Application.java:116)
	at com.shatteredpixel.shatteredpixeldungeon.desktop.DesktopLauncher.main(DesktopLauncher.java:160)
Caused by: java.lang.IndexOutOfBoundsException
	at java.base/java.util.zip.InflaterInputStream.read(InflaterInputStream.java:146)
	at java.base/java.io.FilterInputStream.read(FilterInputStream.java:132)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.getPageAndPacket(OggInputStream.java:178)
	... 11 more

mymedia@barberry:~/код/игры/shattered-pixel-dungeon$ java -jar ./desktop/build/libs/desktop-1.0.2.jar 
[gdx-audio] Error reading OGG: Corrupt or missing data in bitstream.
[GAME] com.badlogic.gdx.utils.GdxRuntimeException: Unable to allocate audio buffers. AL Error: 40963
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALMusic.play(OpenALMusic.java:83)
	at com.watabou.noosa.audio.Music.play(Music.java:169)
	at com.watabou.noosa.audio.Music.access$400(Music.java:36)
	at com.watabou.noosa.audio.Music$1$1.run(Music.java:157)

[GAME] java.lang.NullPointerException
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALLwjgl3Audio.update(OpenALLwjgl3Audio.java:253)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.loop(Lwjgl3Application.java:132)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.<init>(Lwjgl3Application.java:116)
	at com.shatteredpixel.shatteredpixeldungeon.desktop.DesktopLauncher.main(DesktopLauncher.java:160)

mymedia@barberry:~/код/игры/shattered-pixel-dungeon$ java -jar ./desktop/build/libs/desktop-1.0.2.jar 
[gdx-audio] Error reading OGG: Corrupt or missing data in bitstream.
[GAME] java.lang.ArrayIndexOutOfBoundsException: arraycopy: length -158 is negative
	at java.base/java.lang.System.arraycopy(Native Method)
	at com.jcraft.jogg.StreamState.pagein(StreamState.java:256)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.readPCM(OggInputStream.java:319)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:436)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:458)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:474)
	at com.badlogic.gdx.backends.lwjgl3.audio.Ogg$Music.read(Ogg.java:43)
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALMusic.fill(OpenALMusic.java:253)
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALMusic.update(OpenALMusic.java:237)
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALLwjgl3Audio.update(OpenALLwjgl3Audio.java:253)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.loop(Lwjgl3Application.java:132)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.<init>(Lwjgl3Application.java:116)
	at com.shatteredpixel.shatteredpixeldungeon.desktop.DesktopLauncher.main(DesktopLauncher.java:160)

mymedia@barberry:~/код/игры/shattered-pixel-dungeon$ java -jar ./desktop/build/libs/desktop-1.0.2.jar 
[gdx-audio] Error reading OGG: Corrupt or missing data in bitstream.
[GAME] com.badlogic.gdx.utils.GdxRuntimeException: Error during Vorbis decoding.
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.readPCM(OggInputStream.java:404)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:436)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:458)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:474)
	at com.badlogic.gdx.backends.lwjgl3.audio.Ogg$Music.read(Ogg.java:43)
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALMusic.fill(OpenALMusic.java:253)
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALMusic.update(OpenALMusic.java:237)
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALLwjgl3Audio.update(OpenALLwjgl3Audio.java:253)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.loop(Lwjgl3Application.java:132)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.<init>(Lwjgl3Application.java:116)
	at com.shatteredpixel.shatteredpixeldungeon.desktop.DesktopLauncher.main(DesktopLauncher.java:160)
Caused by: java.util.zip.ZipException: invalid distance too far back
	at java.base/java.util.zip.InflaterInputStream.read(InflaterInputStream.java:164)
	at java.base/java.io.FilterInputStream.read(FilterInputStream.java:132)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.readPCM(OggInputStream.java:402)
	... 10 more

mymedia@barberry:~/код/игры/shattered-pixel-dungeon$ java -jar ./desktop/build/libs/desktop-1.0.2.jar 
[GAME] java.lang.NullPointerException
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALLwjgl3Audio.update(OpenALLwjgl3Audio.java:253)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.loop(Lwjgl3Application.java:132)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.<init>(Lwjgl3Application.java:116)
	at com.shatteredpixel.shatteredpixeldungeon.desktop.DesktopLauncher.main(DesktopLauncher.java:160)

mymedia@barberry:~/код/игры/shattered-pixel-dungeon$ java -jar ./desktop/build/libs/desktop-1.0.2.jar 
[gdx-audio] Error reading OGG: Corrupt or missing data in bitstream.
[GAME] com.badlogic.gdx.utils.GdxRuntimeException: Error during Vorbis decoding.
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.readPCM(OggInputStream.java:404)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:436)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:458)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:474)
	at com.badlogic.gdx.backends.lwjgl3.audio.Ogg$Music.read(Ogg.java:43)
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALMusic.fill(OpenALMusic.java:253)
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALMusic.play(OpenALMusic.java:93)
	at com.watabou.noosa.audio.Music.play(Music.java:169)
	at com.watabou.noosa.audio.Music.access$400(Music.java:36)
	at com.watabou.noosa.audio.Music$1$1.run(Music.java:157)
Caused by: java.util.zip.ZipException: invalid distance too far back
	at java.base/java.util.zip.InflaterInputStream.read(InflaterInputStream.java:164)
	at java.base/java.io.FilterInputStream.read(FilterInputStream.java:132)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.readPCM(OggInputStream.java:402)
	... 9 more

[GAME] com.badlogic.gdx.utils.GdxRuntimeException: Error during Vorbis decoding.
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.readPCM(OggInputStream.java:404)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:436)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:458)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.read(OggInputStream.java:474)
	at com.badlogic.gdx.backends.lwjgl3.audio.Ogg$Music.read(Ogg.java:43)
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALMusic.fill(OpenALMusic.java:253)
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALMusic.update(OpenALMusic.java:237)
	at com.badlogic.gdx.backends.lwjgl3.audio.OpenALLwjgl3Audio.update(OpenALLwjgl3Audio.java:253)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.loop(Lwjgl3Application.java:132)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.<init>(Lwjgl3Application.java:116)
	at com.shatteredpixel.shatteredpixeldungeon.desktop.DesktopLauncher.main(DesktopLauncher.java:160)
Caused by: java.util.zip.ZipException: invalid distance too far back
	at java.base/java.util.zip.InflaterInputStream.read(InflaterInputStream.java:164)
	at java.base/java.io.FilterInputStream.read(FilterInputStream.java:132)
	at com.badlogic.gdx.backends.lwjgl3.audio.OggInputStream.readPCM(OggInputStream.java:402)
	... 10 more

mymedia@barberry:~/код/игры/shattered-pixel-dungeon$ git -P log -1
commit e3ffc875ba1cc2abf12293fa3a8ebd94bc527ab0 (HEAD -> master, tag: v1.0.2, origin/master, origin/HEAD)
Author: Evan Debenham <Evan@ShatteredPixel.com>
Date:   2021-08-28 16:04:44 -0400

    v1.0.2: final commit
```

**Comment by Crystal-RainSlide** [comment]
I got this error on Windows 10, which is 2nd one of the errors above:

<details>

<summary>error message</summary>

```
Shattered Pixel Dungeon has run into an error it can't recover from and has crashed, sorry about that!

If you could, please email this error message to the developer (Evan@ShatteredPixel.com):

version: 1.0.2
utils.GdxRuntimeException: Failure reading Vorbis.
    at backends.lwjgl3.audio.OggInputStream.getPageAndPacket(180)
    at backends.lwjgl3.audio.OggInputStream.readPCM(293)
    at backends.lwjgl3.audio.OggInputStream.read(436)
    at backends.lwjgl3.audio.OggInputStream.read(458)
    at backends.lwjgl3.audio.OggInputStream.read(474)
    at backends.lwjgl3.audio.Ogg$Music.read(43)
    at backends.lwjgl3.audio.OpenALMusic.fill(253)
    at backends.lwjgl3.audio.OpenALMusic.update(237)
    at backends.lwjgl3.audio.OpenALLwjgl3Audio.update(253)
    at backends.lwjgl3.Lwjgl3Application.loop(132)
    at backends.lwjgl3.Lwjgl3Application.<init>(116)
    at desktop.DesktopLauncher.main(160)
Caused by: java.lang.IndexOutOfBoundsException
    at java.base/java.util.zip.InflaterInputStream.read(Unknown Source)
    at java.base/java.io.FilterInputStream.read(Unknown Source)
    at backends.lwjgl3.audio.OggInputStream.getPageAndPacket(178)
    ... 11 more
```

Copied from the error popup with [Textify](https://rammichael.com/textify). For the desktop release, you may want to put the error message into a textfield or something like that, so people who close/hide their terminal (like me) can copy from the error popup directly.

</details>

**Comment by 00-Evan** [comment]
I'm aware of the cause for this, it's a race condition that's particular to the desktop audio implementation. Will have it fixed in 1.0.3

**Comment by 00-Evan** [comment]
should be fixed in commit 30c00f95b61493d653e9982077ded5aeb1fc0419

**Comment by mymedia2** [comment]
I can confirm the issue fixed. Thanks!
